akka-stream/src/main/scala/akka/stream/impl/JsonObjectParser.scala

```diff
@@ -36,6 +37,469 @@ import scala.annotation.switch
+
+        case LineBreak ⇒
+          pp.skip()
+          pp.enterState(LinebreakSeparatedBeforeElement)
+
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+
+        /* This is a tolerance; since there is no ambiguity, we can consider a next object immediately after
+          the previous, even without a Comma or Linebreak separator. Akka Stream 2.5.16 has historically supported that.
+
+          We can't extend the same tolerance to arrays, as any stream whose first non-whitespace character is a SquareBraceStart
+          will be considered to be a JSON Array stream
+           */
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterObject, NotArrayAfterElement)
+
+        case _ ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}] — junk after value in linebreak or comma-separated stream")
+      }
+    }
+
+    private abstract sealed class SeparatorSeparatedAfterElement extends ParserState {
+      def separator: Int
+      def separatorName: String
+
+      def beforeNextItem: ParserState
+
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = input match {
+        case _ if input == separator ⇒
+          pp.skip()
+          pp.enterState(beforeNextItem)
+
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+
+        /* This is a tolerance; since there is no ambiguity, we can consider a next object immediately after
+        the previous, even without a Comma or Linebreak separator. Akka Stream 2.5.16 has historically supported that.
+
+        We can't extend the same tolerance to arrays, as any stream whose first non-whitespace character is a SquareBraceStart
+        will be considered to be a JSON Array stream
+         */
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterObject, this)
+
+        case _ ⇒
+          throw new FramingException(s"Invalid JSON encountered at position [${pp.pos}] of [${pp.buffer}] — junk after value in $separatorName-separated stream")
+
+      }
+    }
+
+    private abstract sealed class SeparatorSeparatedBeforeElement extends ParserState {
+      def afterState: SeparatorSeparatedAfterElement
+
+      final override def proceed(input: Byte, pp: JsonObjectParser): Unit = input match {
+        case DoubleQuote ⇒
+          pp.take()
+          pp.enterState(InOuterString(afterState, pp))
+
+        case SquareBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterArray, afterState)
+
+        case CurlyBraceStart ⇒
+          pp.take()
+          pp.enterContainer(InOuterObject, afterState)
+
+        case _ if input == afterState.separator ⇒
+          pp.skip()
+        /* note: effectively, empty lines are tolerated, ignored and skipped. Throw a FramingError if desired otherwise */
+
+        case _ if isWhitespace(input) ⇒
+          pp.skip()
+
+        case _ if !isWhitespace(input) ⇒
```

Comment:
Guard not needed here, the byte is either whitespace and got caught by the above clause or it is not and then it will end up here.

Reply:
stylistic issue. I tend to like this for clarity of symmetry. DRY is a fair opposite. Removing.